File: Formula-Bollo/src/shared/models/driver.ts
import { Season } from "./season";
import { Team } from "./team";

export class Driver {
  id: number;
  name: string;
  number: number;
  team: Team;
  driverImage: string;
  season: Season;

  constructor(
    id: number,
    name: string,
    number: number,
    team: Team,
    driverImage: string,
    season: Season,
  ) {
    this.id = id;
    this.name = name;
    this.number = number;
    this.team = team;
    this.driverImage = driverImage;
    this.season = season;
  }
}

File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/Driver.java
package formula.bollo.app.entity;

import java.sql.Blob;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "driver")
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
public class Driver {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "name", nullable = false, length = 25)
    @NotBlank(message = "El nombre del piloto es obligatorio")
    private String name;

    @Column(name = "number", nullable = false, length = 4)
    @NotBlank(message = "El número de piloto es obligatorio")
    private int number;

    @ManyToOne
    @JoinColumn(name = "team_id", referencedColumnName = "id")
    private Team team;

    @JsonIgnore
    @Column(name = "driver_image", columnDefinition = "BLOB")
    private Blob driverImage;

    @ManyToOne
    @JoinColumn(name = "season_id", referencedColumnName = "id")
    private Season season;
}
